### DISCUSSION TITLE
[BUG] Failed to compile app

### DISCUSSION THREAD

**Discussion by fresher1chen** [alibaba/GraphScope#1135] [Q&A] [answered]
**Describe the bug**
error 

**To Reproduce**
data:p2p-31_property_v_0  p2p-31_property_e_0
refer:https://graphscope.io/docs/v0.7.0/zh/analytics_engine.html#pregel 
运行 上述链接的 pregel 编程模型 的 demo  时，出现以下错误：
![image](https://user-images.githubusercontent.com/44836806/143560857-fac24dc9-a007-4de3-8eeb-406ba6fd921c.png)
在  https://try.graphscope.app/ 平台上  执行时，则可以正常运行。此平台版本 graphscope 为 0.4.请帮忙看看是什么原因导致的，想确认下 0.7、0.8、0.9版本是否还能用 pregel 模型进行分析，感谢。

**Environment (please complete the following information):**
 - GraphScope version: 0.7/0.8
 - OS:Ubuntu
 - Version ：Ubuntu 16.04.3 LTS 
 - Kubernetes Version ：1.22

**Additional context**

**Comment by siyuan0322** [comment]
根据我的经验来说，类似 Killed(program cc1plus) 这种错误，是编译的时候OOM了。coordinator Pod 的内存给的小了，可以调到4G 试试。

**Comment by sighingnow** [comment]
@lidongze0629 Could you please update the configuration of try.graphscope.app to migrate to our latest release?

**Comment by fresher1chen** [comment]
> 根据我的经验来说，类似 Killed(program cc1plus) 这种错误，是编译的时候OOM了。coordinator Pod 的内存给的小了，可以调到4G 试试。

谢谢您，使用 k8s 的成功了，但是使用 0.8 的 单机 版本的时候 还是 出现了问题 。
![image](https://user-images.githubusercontent.com/44836806/143564790-79aecb0c-d74a-4893-bbbf-e2e4bcf173b2.png)

**Comment by siyuan0322** [comment]
看起来是编译时出了错误。报错里应该有一个本地的路径是app的临时编译目录，没有的话你可以找一下 `/tmp/gs/` 下面应该有随机字符串的目录，你找到时间是最近修改的目录，然后进到 `session_xxx/yyyyy` 这种类型的目录，这里面是有makefile的，手动make一下然后将报错贴上来。

**Comment by sighingnow** [comment]
> 根据我的经验来说，类似 Killed(program cc1plus) 这种错误，是编译的时候OOM了。coordinator Pod 的内存给的小了，可以调到4G 试试。

Could we add an options to report the compile error message to the client?

**Comment by lidongze0629** [comment]
> @lidongze0629 Could you please update the configuration of try.graphscope.app to migrate to our latest release?

Going on at v0.9.0

**Comment by fresher1chen** [comment]
> 看起来是编译时出了错误。报错里应该有一个本地的路径是app的临时编译目录，没有的话你可以找一下 `/tmp/gs/` 下面应该有随机字符串的目录，你找到时间是最近修改的目录，然后进到 `session_xxx/yyyyy` 这种类型的目录，这里面是有makefile的，手动make一下然后将报错贴上来。
![image](https://user-images.githubusercontent.com/44836806/143796486-121ca07c-71ba-455b-b338-12b1b23e5bfb.png)
请问一下是这个路径下的Makefile吗？make 结果错误如图所示。

**Comment by lidongze0629** [comment]
@fresher1chen  Does folly exist in your local environment, and could you tell us the version of folly.

**Comment by lidongze0629** [comment, marked as answer]
> > 看起来是编译时出了错误。报错里应该有一个本地的路径是app的临时编译目录，没有的话你可以找一下 `/tmp/gs/` 下面应该有随机字符串的目录，你找到时间是最近修改的目录，然后进到 `session_xxx/yyyyy` 这种类型的目录，这里面是有makefile的，手动make一下然后将报错贴上来。
> > ![image](https://user-images.githubusercontent.com/44836806/143796486-121ca07c-71ba-455b-b338-12b1b23e5bfb.png)
> > 请问一下是这个路径下的Makefile吗？make 结果错误如图所示。

这个问题是由于 folly 在低版本[gcc(5.4)](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=71875)下不兼容造成的，升级gcc (>=7.1.0) 可以解决这个问题 (ubuntu16 + python3.9 + gcc 7.10)

ubuntu16 下升级 gcc可以参考如下链接:

https://askubuntu.com/questions/859256/how-to-install-gcc-7-or-clang-4-0
https://phoenixnap.com/kb/add-apt-repository-command-not-found-ubuntu
https://tuxamito.com/wiki/index.php/Installing_newer_GCC_versions_in_Ubuntu

```shell
sudo apt install software-properties-common

sudo add-apt-repository ppa:ubuntu-toolchain-r/test
sudo apt-get update
sudo apt-get install gcc-7 g++-7
gcc-7 --version

sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-7 70 --slave /usr/bin/g++ g++ /usr/bin/g++-7
sudo update-alternatives --config gcc

gcc -v
```
